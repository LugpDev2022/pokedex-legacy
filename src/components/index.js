export * from './Navbar';
export * from './PokeCard';
