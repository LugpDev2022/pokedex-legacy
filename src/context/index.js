export * from './PokemonContext';
export * from './PokemonProvider';
export * from './pokemonReducer';
export * from './types/types';
