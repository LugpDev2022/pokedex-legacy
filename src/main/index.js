export * from './components';
export * from './hooks';
export * from './MainPage';
