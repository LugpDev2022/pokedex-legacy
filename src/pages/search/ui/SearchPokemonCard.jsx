import pikachu from '../../../assets/images/pikachu.webp';

export const SearchPokemonCard = () => (
  <div className={'col-10 col-sm-8 col-md-6 col-lg-7'}>
    <div className='card card-img-top'>
      <img className='p-2' src={pikachu} alt='Pokemon Image' />
      <div className='card-body shadow p-2'>
        <h3 className='card-title text-center'>Search a Pok√©mon</h3>
      </div>
    </div>
  </div>
);
