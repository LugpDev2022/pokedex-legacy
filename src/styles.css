@font-face {
  font-family: 'pokemonFont';
  src: url('./assets/fonts/pokemonsolid.ttf');
}

.pokemonFont {
  font-family: pokemonFont;
}

body {
  background-color: #d9d9d9;
}
