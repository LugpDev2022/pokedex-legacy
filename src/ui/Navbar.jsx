import { useState } from 'react';

import { Link, useNavigate } from 'react-router-dom';

import { FaSearch } from 'react-icons/fa';
import { BiUser, BiLogIn } from 'react-icons/bi';

export const Navbar = () => {
  const [logged, setLogged] = useState(false);

  const navigate = useNavigate();

  const onLogInClick = () => {
    setLogged(!logged);
  };

  const onClickSearch = () => {
    navigate('/search');
  };

  return (
    <div
      className='navbar navbar-expand-md navbar-light py-3'
      style={{ backgroundColor: '#5986fd' }}
    >
      <div className='container justify-content-between'>
        <Link to='/' className='navbar-brand d-flex align-items-center gap-3'>
          <h1 className='mb-0 pokemonFont'>Pokedex</h1>
        </Link>
        <button
          className='navbar-toggler'
          type='button'
          data-bs-toggle='collapse'
          data-bs-target='#navbarContent'
          aria-controls='navbarContent'
          aria-expanded='false'
          aria-label='Toggle navigation'
        >
          <span className='navbar-toggler-icon'></span>
        </button>

        <div
          className='collapse navbar-collapse justify-content-end'
          id='navbarContent'
        >
          <div className='navbar-nav d-flex align-items-center gap-4'>
            <button onClick={onClickSearch} className='btn'>
              <FaSearch className='d-none d-md-block fs-2' />
              <span className='d-md-none '>Search pokemon</span>
            </button>

            {logged ? (
              <button
                onClick={onLogInClick}
                className='btn btn-outline-dark d-flex align-items-center gap-2'
              >
                <BiUser className='fs-1' />
                <span className='fs-4'>Account</span>
              </button>
            ) : (
              <button
                onClick={onLogInClick}
                className='btn btn-outline-dark d-flex align-items-center gap-2'
              >
                <BiLogIn className='fs-2' />
                <span className='fs-4'>Log In</span>
              </button>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
