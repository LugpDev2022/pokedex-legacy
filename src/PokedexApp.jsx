import { ContextProvider } from './context';
import { AppRouter } from './router';
import { Navbar } from './ui';

const PokedexApp = () => (
  <ContextProvider>
    <Navbar />
    <AppRouter />
  </ContextProvider>
);

export default PokedexApp;
