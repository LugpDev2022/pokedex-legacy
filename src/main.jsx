import React from 'react';
import ReactDOM from 'react-dom/client';

import * as bootstrap from 'bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

import PokedexApp from './PokedexApp';
import { BrowserRouter } from 'react-router-dom';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <BrowserRouter>
      <PokedexApp />
    </BrowserRouter>
  </React.StrictMode>
);
