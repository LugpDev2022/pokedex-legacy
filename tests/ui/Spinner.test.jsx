import { render } from '@testing-library/react';
import { Spinner } from '../../src/ui';

describe('Tests on <Spinner />', () => {
  test('should match with snapshot', () => {
    const { container } = render(<Spinner />);

    expect(container).toMatchSnapshot();
  });
});
