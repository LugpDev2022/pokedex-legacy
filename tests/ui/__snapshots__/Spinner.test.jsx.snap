// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Spinner /> should match with snapshot 1`] = `
<div>
  <div
    class="position-absolute top-50 start-50 translate-middle"
  >
    <div
      class="spinner-border text-primary"
      role="status"
    >
      <span
        class="visually-hidden"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
